axelor-base/src/main/java/com/axelor/apps/base/web/CountryController.java

```diff
@@ -0,0 +1,35 @@
+package com.axelor.apps.base.web;
+
+import com.axelor.apps.base.service.CountryService;
+import com.axelor.inject.Beans;
+import com.axelor.rpc.ActionRequest;
+import com.axelor.rpc.ActionResponse;
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CountryController {
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  public void recomputeAllAddress(ActionRequest request, ActionResponse response) {
+    CountryService countryService = Beans.get(CountryService.class);
+    Object ids = request.getContext().get("_ids");
+
+    if (ids == null) {
+      response.setInfo("Needs to select at least one country");
+    } else {
+      List<Long> countryIdsList = (List<Long>) ids;
+      Pair<Integer, Integer> pair = countryService.recomputeAllAddress(countryIdsList);
```

Comment:
Addresses formattedFullName is never changed, maybe it's because there is no save in AddressService.setFormattedFullName